Sources/SyncDataProviders/Bookmarks/internal/BookmarksResponseHandler.swift

```diff
@@ -113,33 +113,38 @@ final class BookmarksResponseHandler {
         }
         try processOrphanedBookmarks()
 
-        // populate favorites
-        if !favoritesUUIDs.isEmpty {
-            guard let favoritesFolder = BookmarkUtils.fetchFavoritesFolder(context) else {
-                // Error - unable to process favorites
-                return
-            }
+        processReceivedFavorites()
+    }
 
-            // For non-first sync we rely fully on the server response
-            if !shouldDeduplicateEntities {
-                favoritesFolder.favoritesArray.forEach { $0.removeFromFavorites() }
-            } else if !favoritesFolder.favoritesArray.isEmpty {
-                // If we're deduplicating and there are favorires locally, we'll need to sync favorites folder back later.
-                // Let's keep its modifiedAt.
-                idsOfItemsThatRetainModifiedAt.insert(BookmarkEntity.Constants.favoritesFolderID)
-            }
+    // MARK: - Private
 
-            favoritesUUIDs.forEach { uuid in
-                if let bookmark = entitiesByUUID[uuid] {
-                    bookmark.removeFromFavorites()
-                    bookmark.addToFavorites(favoritesRoot: favoritesFolder)
-                }
+    private func processReceivedFavorites() {
+        guard let favoritesUUIDs else {
+            return
+        }
+
+        guard let favoritesFolder = BookmarkUtils.fetchFavoritesFolder(context) else {
+            // Error - unable to process favorites
+            return
+        }
+
+        // For non-first sync we rely fully on the server response
+        if !shouldDeduplicateEntities {
+            favoritesFolder.favoritesArray.forEach { $0.removeFromFavorites() }
```

Comment:
As per call - we will need to validate local timestamp here.

Reply:
Added a task [here](https://app.asana.com/0/1201493110486074/1205896860658963/f).